main.go

```diff
@@ -74,3 +77,91 @@ func main() {
 	}()
 	s.ListenAndServe()
 }
+
```

Comment:
These methods (and the associated test suite -- which is good) feel out of place here.

We had talked about cleaning up all the miscellaneous files & methods at the top of the project.

I don't think these belong under `pkg` and exposed publicly. Is this the opportunity to start using an `internal` area and getting some non-public packages setup there?

Options & configuration validation logic jump to my mind as good candidates for an `internal` placement.

Reply:
I'd happily move them to another file if you'd prefer, but I think making a package for this might end up being overkill as this should all go away in our long term plan right? Once we are structured config only this can all be deleted.

I'm not sure about the use of `internal`, how much do we care if other people are importing these bits of code? We aren't claiming this is a library, but if people want to use bits of the code I don't really mind (eg middleware for example). I could actually imagine someone importing the `Options` package to generate configuration or something so we probably don't want to make that internal either.

I feel like we could possibly have a big conversation about this 😅  I might read up on when people are using `internal`